api/v1/gatewayapi_types.go

```diff
@@ -22,6 +22,14 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// +kubebuilder:validation:Enum=Reconcile;PreferExisting
+type GatewayCRDManagement string
+
+const (
+	GatewayCRDManagementReconcile      GatewayCRDManagement = "Reconcile"
```

Comment:
We could probably use this terminology / setting for any third part crd management, i.e. take away the `Gateway` prefix and put this in a common folder.